Tests/ReactiveSwiftTests/PropertySpec.swift

```diff
@@ -819,6 +819,64 @@ class PropertySpec: QuickSpec {
 				}
 			}
 
+			describe("lens") {
+				it("should return the result of the getter on each value change") {
+					let initialValue = (character: "🎃", other: 42)
+					let nextValue = (character: "😾", other: 74)
+
+					let tupleProperty = MutableProperty<(character: String, other: Int)>(initialValue)
+					let theLens = tupleProperty.lens { $0.character }
+
+					expect(theLens).to(sendValue("🎃", sendError: nil, complete: false))
+
+					tupleProperty.value = nextValue
+
+					expect(theLens).to(sendValue("😾", sendError: nil, complete: false))
+				}
+
+				it("should evaluate its getter lazily") {
+					let initialValue = (character: "🎃", other: 42)
+					let tupleProperty = MutableProperty<(character: String, other: Int)>(initialValue)
+
+					var getterEvaluated = false
+					let theLens = tupleProperty.lens { (tuple: (character: String, other: Int)) -> String in
+						getterEvaluated = true
+						return tuple.character
+					}
+
+					expect(getterEvaluated).to(beFalse())
+					expect(theLens).to(sendValue("🎃", sendError: nil, complete: false))
+					expect(getterEvaluated).to(beTrue())
+				}
+
+				it("should evaluate its getter lazily on the specified scheduler") {
+					let initialValue = (character: "🎃", other: 42)
+					let tupleProperty = MutableProperty<(character: String, other: Int)>(initialValue)
+
+					let testScheduler = TestScheduler()
+
+					var getterEvaluated = false
+					let theLens = tupleProperty.lens(on: testScheduler) { (tuple: (character: String, other: Int)) -> String in
+						getterEvaluated = true
+						return tuple.character
+					}
+
+					var characters: [String] = []
+					theLens.startWithValues { character in
+						characters.append(character)
+					}
+
+					expect(getterEvaluated).to(beFalse())
+					expect(characters).to(beEmpty())
+
+					testScheduler.run()
```

Comment:
The test I added merely intended to demonstrate the following:

* Before the signal is started, the getter has not been evaluated
* When the signal is started, but the getter's queue hasn't scheduled the getter for execution, the getter has still not been evaluated
* Once the getter's queue is scheduled, the setter will have run, and the value will have been delivered

I'm probably missing the false positive here, but I think the synchronous execution of the scheduler is a benefit for the specific (simple!) test case I had in mind. Namely, "will the getter get scheduled at all, rather than executing immediately when the `SignalProducer` is started.."

Now, what you're proposing could help write another test case I wasn't feeling well-equipped to write, which is, "the getter should execute on the specific scheduler/thread that is specified".

Let me know what I might have misinterpreted.